src/test/java/net/unicon/lti/utils/lti/LTI3RequestTest.java

```diff
@@ -183,26 +193,138 @@ public void testLTI3RequestWithInvalidNonce() {
         verify(claims).get(eq(LtiStrings.LTI_VERSION), eq(String.class));
         verify(claims).get(eq(LtiStrings.LTI_MESSAGE_TYPE), eq(String.class));
         verify(claims).get(eq(LtiStrings.LTI_NONCE), eq(String.class));
-        assertEquals("Nonce error: Unknown or already used nonce.", exception.getMessage());
+        assertEquals("Nonce error: Nonce not found in the database.", exception.getMessage());
     }
 
     @Test
     public void testLTI3RequestWithValidNonce() {
+        Map<String,Object> testMap = new HashMap<>();
+        testMap.put("id","value1");
+        List<String> testList = new ArrayList<>();
+        testList.add("key1");
+        when(req.getParameter("id_token")).thenReturn(ID_TOKEN);
```

Comment:
I think the tests might be more readable if all of the common `when(claims...).thenReturn()` code is moved into the `setUp` method if that makes sense to do with the rest of these tests.

Reply:
Moved some of those to the top.